### ISSUE TITLE
[Bug] --config option always looks in CWD

### ISSUE THREAD

**Issue by dmwilcox** [grafana/grafana#4531]
* What grafana version are you using?
Built from master yesterday.

* What datasource are you using?
<haven't gotten this far>

* What OS are you running grafana on?
debian 8

I was running into this issue while trying to use --config and environmental variables to set the grafana config location inside a container.  After some fussing I decided to test it outside of the container and realized I found some unexpected behavior.

given a directory structure like:
GR_PKG=$HOME/go/src/github.com/grafana/grafana/ # pkg contents in here

Executing from one directory above $GF_PKG:
./grafana/bin/grafana-server --config grafana/conf/defaults.ini
log.go: 84 FATAL ... Failed to parse defaults.ini, open /home/user/go/src/github.com/grafana/conf/defaults.ini

note the lack of the extra 'grafana directory in the path... there should be two

Note: I also tried ./granfana/conf/defaults.ini and $PWD/grafana/conf/defaults to see if there were any differences given an absolute path (no)

I also tried to execute from my home directory to see how many directory levels it was removing:

whilst still at $GF_PKG/..
$ CONF=$PWD/grafana/conf/defaults.ini
cd $HOME
./$GO_PKG/bin/grafana-server --config $CONF
... Failed to parse defaults.ini, open /home/user/conf/defaults.ini

TLDR; it looks like --config ignores the path given and always looks in the current directory that grafana is launched from.  Launched from the same directory as that containing conf/, data/ and public/ everything appears to work.

This is workable for my given I'm dealing with a container, but similar behavior from public is proving much harder to deal with.  I'll file a bug for that if I can confirm it is something with the path processing and not my configuration.  Thank you.

**Comment by torkelo** [comment]
So this is how it is started from the init d script. 

DAEMON_OPTS="--pidfile=${PID_FILE} --config=${CONF_FILE} cfg:default.paths.data=${DATA_DIR} cfg:default.paths.logs=${LOG_DIR} cfg:default.paths.plugins=${PLUGINS_DIR}"

The PID and LOG/DATA/PLUGIN dirs are optional, if you are ok with the defaults. 

Working dir should be relative to the public directory  (Frontend assets), or you will have to change static_root_path config as well.
